Description: Use system's groovy instead of the embedded one
 .
Author: Balint Reczey <balint@balintreczey.hu>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- xbmc-12.2+dfsg1.orig/codegenerator.mk
+++ xbmc-12.2+dfsg1/codegenerator.mk
@@ -21,7 +19,6 @@ DOXY_XML_PATH=$(GENDIR)/doxygenxml
 endif
 
 GENDIR = $(INTERFACES_DIR)/python/generated
-GROOVY_DIR = $(TOPDIR)/lib/groovy
 
 GENERATED =  $(GENDIR)/AddonModuleXbmc.cpp
 GENERATED += $(GENDIR)/AddonModuleXbmcgui.cpp
@@ -32,8 +31,10 @@ GENERATE_DEPS += $(TOPDIR)/xbmc/interfaces/legacy/*.h
 vpath %.i $(INTERFACES_DIR)/swig
 
 $(GENDIR)/%.cpp: $(GENDIR)/%.xml $(JAVA) $(SWIG) $(DOXY_XML_PATH)
-	$(JAVA) -cp "$(GROOVY_DIR)/groovy-all-1.8.4.jar:$(GROOVY_DIR)/commons-lang-2.6.jar:$(TOPDIR)/tools/codegenerator:$(INTERFACES_DIR)/python" \
-          groovy.ui.GroovyMain $(TOPDIR)/tools/codegenerator/Generator.groovy $< $(INTERFACES_DIR)/python/PythonSwig.cpp.template $@ $(DOXY_XML_PATH)
+	# work around groovy crash
+	groovyc -cp "/usr/share/java/groovy-all-1.8.6.jar:/usr/share/java/commons-lang.jar:./tools/codegenerator:xbmc/interfaces/python" -d tools/codegenerator tools/codegenerator/Helper.groovy  tools/codegenerator/SwigTypeParser.groovy xbmc/interfaces/python/MethodType.groovy xbmc/interfaces/python/PythonTools.groovy
+	groovy -cp "/usr/share/java/groovy-all-1.8.6.jar:/usr/share/java/commons-lang.jar:$(TOPDIR)/tools/codegenerator:$(INTERFACES_DIR)/python" \
+		$(TOPDIR)/tools/codegenerator/Generator.groovy $< $(INTERFACES_DIR)/python/PythonSwig.cpp.template $@ $(DOXY_XML_PATH)
 	rm $<
 
 $(GENDIR)/%.xml: %.i $(SWIG) $(JAVA) $(GENERATE_DEPS)
